javers-core/src/main/java/org/javers/repository/jql/QueryBuilder.java

```diff
@@ -0,0 +1,103 @@
+package org.javers.repository.jql;
+
+import org.javers.common.exception.JaversException;
+import org.javers.common.exception.JaversExceptionCode;
+import org.javers.common.validation.Validate;
+import org.javers.core.Javers;
+import org.javers.core.metamodel.object.CdoSnapshot;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+import static org.javers.repository.jql.InstanceIdDTO.instanceId;
+
+/**
+ * Fluent API for building {@link JqlQuery},
+ * executed with {@link Javers#findChanges(JqlQuery)} and {@link Javers#findSnapshots(JqlQuery)}
+ *
+ * @see Javers#findChanges(JqlQuery)
+ * @author bartosz.walacik
+ */
+public class QueryBuilder {
+    private static int DEFAULT_LIMIT = 100;
+
+    private int limit = DEFAULT_LIMIT;
+    private boolean newObjectChanges;
+    private final List<Filter> filters = new ArrayList<>();
+
+    private QueryBuilder(Filter initialFilter) {
+        addFilter(initialFilter);
+    }
+
+    public static QueryBuilder byClass(Class requiredClass){
+        return new QueryBuilder(new ClassFilter(requiredClass));
+    }
+
+    public static QueryBuilder byInstanceId(Object localId, Class entityClass){
+        Validate.argumentsAreNotNull(localId, entityClass);
+        return new QueryBuilder(new IdFilter(instanceId(localId, entityClass)));
+    }
+
+    public static QueryBuilder byValueObject(Class ownerEntityClass, String path){
+        Validate.argumentsAreNotNull(ownerEntityClass, path);
+        return new QueryBuilder(new VoOwnerFilter(ownerEntityClass, path));
+    }
+
+    public static QueryBuilder byValueObjectId(Object ownerLocalId, Class ownerEntityClass, String path){
+        Validate.argumentsAreNotNull(ownerEntityClass, ownerLocalId, path);
+        return new QueryBuilder(new IdFilter(ValueObjectIdDTO.valueObjectId(ownerLocalId, ownerEntityClass, path)));
+    }
+
+    @Deprecated
+    public static QueryBuilder byGlobalIdDTO(GlobalIdDTO globalId){
+        Validate.argumentIsNotNull(globalId);
+        return new QueryBuilder(new IdFilter(globalId));
+    }
+
+    /**
+     * Selects snapshots with a given property on changed properties list.
+     *
+     * @see CdoSnapshot#getChanged()
+     */
+    public QueryBuilder andProperty(String propertyName) {
+        Validate.argumentIsNotNull(propertyName);
+        addFilter(new PropertyFilter(propertyName));
+        return this;
+    }
+
+    public QueryBuilder withNewObjectChanges(){
+        this.newObjectChanges = true;
```

Comment:
to się troche dziwnie zachowuje, spodziewałem się że jak jest na false to nie będe widział zmian na propertiesach z null -> some value